samples/cpp/cvtColor_RGB2YUV422.cpp

```diff
@@ -0,0 +1,72 @@
+#include "opencv2/core/utility.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/imgcodecs.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/core/ocl.hpp"
+
+#include <stdio.h>
+
+using namespace cv;
+using namespace std;
+
+
+const char* window_name1 = "Original Image";
+const char* window_name2 = "Reconverted image";
+
+static void help(const char** argv)
+{
+    printf("\nThis sample demonstrates conversion from RGB to YUV422 (UYVY) and back.\n"
+           "Call:\n"
+           "    %s [image_name -- Default is fruits.jpg]\n\n", argv[0]);
+}
+
+const char* keys =
+{
+    "{help h||}{@image |fruits.jpg|input image name}"
+};
+
+int main( int argc, const char** argv )
+{
+    help(argv);
+    CommandLineParser parser(argc, argv, keys);
+    string filename = parser.get<string>(0);
+
+    // Whether to use OpenCL
+    cv::ocl::setUseOpenCL(false);
+
+    UMat image, image_yuv, image_yuv2bgr;
+
+    Mat orig_img = imread(samples::findFile(filename), IMREAD_COLOR);
+    if(orig_img.empty())
+    {
+        printf("Cannot read image file: %s\n", filename.c_str());
+        help(argv);
+        return -1;
+    }
+
+    image = orig_img.getUMat(ACCESS_READ);
```

Comment:
Basic samples don't use `UMat` in general. Just `Mat` is enough.

`UMat` samples with OpenCL mode switching should go to `samples/tapi` location.